ui/src/app/applications/components/application-create-panel/application-create-setfinalizer.tsx

```diff
@@ -0,0 +1,32 @@
+import {Checkbox, HelpIcon} from 'argo-ui';
+import * as React from 'react';
+import * as ReactForm from 'react-form';
+
+export const SetFinalizerOnApplication = ReactForm.FormField((props: {fieldApi: ReactForm.FieldApi}) => {
+    const {
+        fieldApi: {getValue, setValue}
+    } = props;
+    const finalizerval = 'resources-finalizer.argocd.argoproj.io';
+    const setval = getValue() || [];
+    const isChecked = setval.indexOf(finalizerval) === -1 ? false : true;
```

Comment:
One more, there is a PR check failing.  I think that line should be something like this:

```
    const index = setval.findIndex((item: string) => (item === finalizerval));
    const isChecked = index < 0 ? false : true;
```

- note to use lower camel case for naming variables

Reply:
let me 
try ur suggestion